### PR TITLE
Return only sharp capable devices

### PR DESCRIPTION
**Pull request by bureddy**

### CODE REVIEW

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 15-Jan-2020

Artefact file | Status
-- | --
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/7748a41ce1ef395aa8d65c7e317a4cea/raw/d6b28629b57176deec5aaf29f099d8deb40ea15f/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/7748a41ce1ef395aa8d65c7e317a4cea/raw/96d0619df2cc4eaf8eaddea19a5b1d5de5bb28ff/configure_sharp_startup.log) | :heavy_check_mark: PASS
[configure_sharp_stop.log](https://gist.githubusercontent.com/mellanox-github/7748a41ce1ef395aa8d65c7e317a4cea/raw/c6172740c16b8d871e95e411d83a14ae6168cb14/configure_sharp_stop.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/7748a41ce1ef395aa8d65c7e317a4cea/raw/1657bf81c069f5c59d2856da1082e2df6532d44d/run_nccl_test.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

File: src/plugin.c [hunk 1]
```diff
@@ -152,6 +166,16 @@ ncclResult_t ncclIbInit(ncclDebugLogger_t logFunction) {
           ncclIbDevs[ncclNIbDevs].link = portAttr.link_layer;
           ncclIbDevs[ncclNIbDevs].context = context;
           strncpy(ncclIbDevs[ncclNIbDevs].devName, devices[d]->name, MAXNAMESIZE);
+          readFileNumber(&vendorId, IB_DEVICE_SYSFS_FMT, devices[d]->name, "vendor");
+          readFileNumber(&devId, IB_DEVICE_SYSFS_FMT, devices[d]->name, "device");
+          ncclIbDevs[ncclNIbDevs].isSharpDev = 0;
+          if ((portAttr.link_layer == IBV_LINK_LAYER_INFINIBAND) &&
+              (vendorId == 0x15b3) &&           // Mellanox verndor
```
Comment (sjeaugey) [inline src/plugin.c]: verndor -> vendor

**Comment by sjeaugey** [comment]
Otherwise looks ok to me.

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 17-Jan-2020

Artefact file | Status
-- | --
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/28b36357a8ba4b524bf01c386d22ecf1/raw/76f7dcd857fe0eda4504303d0f7741cca5d66973/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/28b36357a8ba4b524bf01c386d22ecf1/raw/d2431cbf6d097d3f4511843d1fb405471c2262c5/configure_sharp_startup.log) | :heavy_check_mark: PASS
[configure_sharp_stop.log](https://gist.githubusercontent.com/mellanox-github/28b36357a8ba4b524bf01c386d22ecf1/raw/82682b788af5a881da148c7885c97f1f1ad6a9e5/configure_sharp_stop.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/28b36357a8ba4b524bf01c386d22ecf1/raw/2ab68057f3b6774e877aad0fbc1d6686004b3d9e/run_nccl_test.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

**Pull request merged**